cmd/iq.go

```diff
@@ -0,0 +1,230 @@
+		panic(err)
+	}
+
+	var purls = mod.ExtractPurlsFromManifest()
+
+	err = auditWithIQServer(purls, configIQ.IQApplication)
+	if err != nil {
+		if errExit, ok := err.(customerrors.ErrorExit); ok {
+			os.Exit(errExit.ExitCode)
+		} else {
+			logLady.WithError(err).Error("unexpected error in iq cmd")
+			panic(err)
+		}
+	}
+
+	return
+}
+
+const (
+	flagNameIqUsername    = "iq-username"
+	flagNameIqToken       = "iq-token"
+	flagNameIqStage       = "iq-stage"
+	flagNameIqApplication = "iq-application"
+	flagNameIqServerUrl   = "iq-server-url"
+)
+
+func init() {
+	cobra.OnInitialize(initIQConfig)
+
+	iqCmd.Flags().StringVarP(&configIQ.IQUsername, flagNameIqUsername, "l", "admin", "Specify Nexus IQ username for request")
+	iqCmd.Flags().StringVarP(&configIQ.IQToken, flagNameIqToken, "p", "admin123", "Specify Nexus IQ token for request")
+	iqCmd.Flags().StringVarP(&configIQ.IQStage, flagNameIqStage, "s", "develop", "Specify Nexus IQ stage for request")
+
+	iqCmd.Flags().StringVarP(&configIQ.IQApplication, flagNameIqApplication, "a", "", "Specify Nexus IQ public application ID for request")
+	if err := iqCmd.MarkFlagRequired(flagNameIqApplication); err != nil {
+		panic(err)
+	}
+
+	iqCmd.Flags().StringVarP(&configIQ.IQServer, flagNameIqServerUrl, "x", "http://localhost:8070", "Specify Nexus IQ server url for request")
+
+	rootCmd.AddCommand(iqCmd)
+}
+
+func bindViperIq(cmd *cobra.Command) {
+	// need to defer bind call until command is run. see: https://github.com/spf13/viper/issues/233
+
+	// need to ensure ossi CLI flags will override ossi config file values when running IQ command
+	bindViper(cmd)
+
+	// Bind viper to the flags passed in via the command line, so it will override config from file
+	if err := viper.BindPFlag(configuration.YamlKeyIQUsername, cmd.Flags().Lookup(flagNameIqUsername)); err != nil {
+		panic(err)
+	}
+	if err := viper.BindPFlag(configuration.YamlKeyIQToken, cmd.Flags().Lookup(flagNameIqToken)); err != nil {
+		panic(err)
+	}
+	if err := viper.BindPFlag(configuration.YamlKeyIQServer, cmd.Flags().Lookup(flagNameIqServerUrl)); err != nil {
+		panic(err)
+	}
+}
+
+func initIQConfig() {
+	var cfgFileToCheck string
+	if cfgFileIQ != "" {
+		viper.SetConfigFile(cfgFileIQ)
+		viper.SetConfigType(configTypeYaml)
+		cfgFileToCheck = cfgFileIQ
+	} else {
+		home, err := homedir.Dir()
+		if err != nil {
+			fmt.Println(err)
+			os.Exit(1)
+		}
+		configPath := path.Join(home, types.IQServerDirName)
+
+		viper.AddConfigPath(configPath)
+		viper.SetConfigType(configTypeYaml)
+		viper.SetConfigName(types.IQServerConfigFileName)
+
+		cfgFileToCheck = path.Join(configPath, types.IQServerConfigFileName)
+	}
+
+	if fileExists(cfgFileToCheck) {
+		// 'merge' IQ config here, since we also need OSSI config, and load order is not guaranteed
+		if err := viper.MergeInConfig(); err != nil {
+			panic(err)
+		}
+	}
+}
+
+func auditWithIQServer(purls []string, applicationID string) error {
```

Comment:
Unrelated to this (just making an observation), for `Options` on the IQServer `go-sona-types` struct, I created an `Application` string, which I had sort of intended to allow us to set that in the `Options` rather than pass it all over the place. HOWEVER, I never used it in `AuditPackages` etc..., so you know, wishful thinking. I can likely get back to `go-sona-types` and implement it so that it's used.

Reply:
> Unrelated to this (just making an observation), for `Options` on the IQServer `go-sona-types` struct, I created an `Application` string, which I had sort of intended to allow us to set that in the `Options` rather than pass it all over the place. HOWEVER, I never used it in `AuditPackages` etc..., so you know, wishful thinking. I can likely get back to `go-sona-types` and implement it so that it's used.

👍 Might be "Issue worthy" to help remember todo it.